src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -124,6 +127,86 @@ protected function json($data, $status = 200, $headers = array(), $context = arr
     }
 
     /**
+     * Returns a BinaryFileResponse object with original or customized file name and disposition header.
+     *
+     * @param File|string $file        File object, path to file or string with content to be sent as response
+     * @param null        $fileName    File name to be sent to response or null (will use original file name)
+     * @param string      $disposition Disposition of response (attachment is default, other type is inline)
+     *
+     * @return BinaryFileResponse
+     *
+     * @throws \LogicException|\InvalidArgumentException|\Exception
+     */
+    protected function file($file, $fileName = null, $disposition = ResponseHeaderBag::DISPOSITION_ATTACHMENT)
+    {
+        $deleteFileAfterSend = false;
+
+        // Test if path to file is given
+        if (is_string($file) && file_exists($file)) {
+            $file = new File($file);
+        }
+
+        // Test if content in string is given
+        if (is_string($file)) {
+            if (empty($file)) {
+                throw new \InvalidArgumentException('File content can\'t be empty.');
+            }
+
+            if (empty($fileName)) {
+                throw new \InvalidArgumentException('File name can\'t be empty.');
+            }
+            $cacheDir = $this->container->get('kernel')->getCacheDir();
+            $tmpName = md5($fileName);
+            $tmpPath = $cacheDir.DIRECTORY_SEPARATOR.$tmpName;
+
+            try {
+                file_put_contents($tmpPath, $file);
+            } catch (\Exception $e) {
+                throw $e;
+            }
+
+            $deleteFileAfterSend = true;
+            $file = new File($tmpPath);
+        }
+
+        if ($file instanceof File && $file->isReadable()) {
+            $response = new BinaryFileResponse($file);
+            $mimeType = $file->getMimeType();
+            if ($deleteFileAfterSend === true) {
+                $response->deleteFileAfterSend($deleteFileAfterSend);
+            }
+        } else {
+            throw new \InvalidArgumentException(
```

Comment:
@HeahDude Well, it's hard for me to decide which solution is better. I thought that passing value would be more clear when reading error message, huh?